### DISCUSSION TITLE
Pass a string to xarray.Dataset.reduce

### DISCUSSION THREAD

**Discussion by tqa236** [pydata/xarray#7537] [Q&A] [answered]
Hello,

I'm not sure if this feature exists or not, but I would like to know if it's possible to pass a string to `xarray.Dataset.reduce` as the aggregation method and receive the corresponding reduced dataset instead of a series of if else like this

```
def aggregate(dataset, aggregation_method):
    if aggregation_method == "mean":
        return dataset.mean()
    elif aggregation_method == "median":
        return dataset.median()
    elif aggregation_method == "max":
        return dataset.max()
```

From the [docstring](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.reduce.html), it seems that the method only accept `Callable` for `func`

Thank you very much.

**Comment by keewis** [comment, marked as answer]
additional keyword-arguments are passed through (this is what `**kwargs` in the docstring is referring to):
```python
ds.reduce(aggregate, aggregation_method="mean")
```

you could also use `functools.partial` or a `lambda` to create a function that has that parameter pre-set:
```python
ds.reduce(functools.partial(aggregate, aggregation_method="mean"))
ds.reduce(lambda ds_: aggregate(ds_, "mean"))
```

Edit: if `aggregation_method` is the name of a method on the dataset, we can also do:
```python
getattr(ds, aggregation_method)()
```

**Reply by tqa236** [reply]
Hi @keewis , 

Thanks for your reply. I probably haven't phrased my question quite well. I would like to know if it's possible to do something like `ds.reduce("mean")` without the need to define the function `aggregate`. 

Something like this (this code doesn't work)

```python
ds.reduce(getattr(ds, aggregation_method))
```

**Reply by keewis** [reply]
for this we can make use of `python`'s bound method objects:
```python
getattr(ds, aggregation_method)()
```
If `aggregation_method` is the name of a method, `getattr` returns a function with `ds` bound to the first argument (usually named `self` by convention), so we can just call that.

**Reply by tqa236** [reply]
thank you very much for your help
